.ci/create_icub-main_deb.sh

```diff
@@ -222,45 +228,175 @@ check_and_install_deps()
+
+  if [ "$?" != "0" ]; then
+    echo "Error: unable to install ycm"
+    exit 1
+  fi
+
+###----------- Handle iCub deps from icub-common -------###
+  echo "Installing icub-common dependencies in the environment"
+  DEP_TAG="ICUB_DEPS_${_PLATFORM_RELEASE}"
+  _DEPENDENCIES="$ICUB_DEPS_COMMON ${!DEP_TAG}"
+  DEBIAN_FRONTEND=noninteractive; $_SUDO apt-get install $APT_OPTIONS $_DEPENDENCIES
+
+  if [ "$?" != "0" ]; then
+    echo "Error: unable to install dependencies"
+    exit 1
+  fi
+
+###------------------- Handle IpOpt --------------------###
+## It seems already handeled by $_SUDO apt install coinor-libipopt-dev
+
+###------------------- Handle YARP --------------------###
+  echo "Installing YARP package"
+  YARP_URL_TAG="YARP_PACKAGE_URL_${_PLATFORM_RELEASE}"
+  wget ${!YARP_URL_TAG} -O /tmp/yarp.deb
+  DEBIAN_FRONTEND=noninteractive; $_SUDO dpkg -i /tmp/yarp.deb
+
+  if [ "$?" != "0" ]; then
+    echo "Error: unable to install yarp"
+    exit 1
+  fi
+}
+
+build_icub() {
+
+  echo "Cloning icub sources from ${ICUB_REPO_URL}"
+  git clone $ICUB_REPO_URL
+  if [ "$?" != "0" ]; then
+    echo "Error: unable to clone icub repositoy from ${ICUB_REPO_URL}"
+    exit 1
+  fi
+  cd icub-main
+  export D_ICUB_ROOT=$(pwd)
+  git checkout v$ICUB_PACKAGE_VERSION
+  if [ "$?" != "0" ]; then
+    echo "Error: unable to checkout to v$ICUB_PACKAGE_VERSION"
+    exit 1
+  fi
+
+  mkdir build && cd build
+
+  CMAKE_OPTIONS_TAG="CMAKE_OPTIONS_${PLATFORM_KEY}"
+  _SPECIAL_DIST_CMAKE_OPTIONS="${!CMAKE_OPTIONS_TAG}"
+  export D_ICUB_INSTALL_DIR=$(pwd)/install
+  cmake $ICUB_CMAKE_OPTIONS -DCMAKE_INSTALL_PREFIX=$D_ICUB_INSTALL_DIR/usr ..
+
+  make -j && make install
+  if [ "$?" != "0" ]; then
+    echo "Error: unable to checkout to build and/or install icub-main"
+    exit 1
+  fi
+}
+
+fix_relocatable_files(){
+  ICUB_INI_PATH="$D_ICUB_INSTALL_DIR/usr/share/yarp/config/path.d"
+  ICUB_INI_FILE="iCub.ini"
+
+  echo "Fixing iCub.ini in path.d"
+  # this fixes missing iCub.ini file
+  if [ ! -e "${ICUB_INI_PATH}/${ICUB_INI_FILE}" ]
+  then
+    mkdir -p $ICUB_INI_PATH
+    ls  $ICUB_INI_PATH
+    touch ${ICUB_INI_PATH}/${ICUB_INI_FILE}
+    echo ###### This file is automatically generated by CMake. >> ${ICUB_INI_PATH}/${ICUB_INI_FILE}
+    echo [search iCub] >> ${ICUB_INI_PATH}/${ICUB_INI_FILE}
+    echo path "/usr/share/iCub">> ${ICUB_INI_PATH}/${ICUB_INI_FILE}
+  fi
+  echo "Fix path inside cmake files"
+  _cmake_files=$(find ${D_ICUB_INSTALL_DIR} -name "*.cmake")
+  for f in $_cmake_files ; do
+    sed -i "s|$D_ICUB_INSTALL_DIR||g" $f
+  done
+
+  echo "Fix path inside  ini files"
+  _ini_files=$(find ${D_ICUB_INSTALL_DIR} -name "*.ini")
+  for f in $_ini_files ; do
+    sed -i "s|$D_ICUB_INSTALL_DIR||g" $f
+  done
+
 }
 
 main()
 {
-  create_control_file
+  install_deps
+  build_icub
+  # Probably this step is not needed anymore after cmake components/modernization
+  fix_relocatable_files
```

Comment:
I think it is still necessary for: 
* https://github.com/robotology/yarp/issues/2445
* https://github.com/robotology/yarp/issues/2444